edit/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+use std::ffi::OsString;
+use std::str::FromStr;
+
+use radicle_common::args::{Args, Error, Help};
+use radicle_common::{keys, profile};
+use radicle_terminal as term;
+
+use librad::git::identities::{any, person, project, SomeIdentity};
+use librad::git::Urn;
+
+use link_identities::payload::{PersonPayload, ProjectPayload};
+
+use anyhow::anyhow;
+
+pub const HELP: Help = Help {
+    name: "edit",
+    description: env!("CARGO_PKG_DESCRIPTION"),
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad edit [<urn>] [<option>...]
+
+    Edits the identity document pointed to by the URN. If it isn't specified,
+    the current project is edited.
+
+Options
+
+    --help              Print help
+"#,
+};
+
+#[derive(Default, Debug, Eq, PartialEq)]
+pub struct Options {
+    pub urn: Option<Urn>,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut urn: Option<Urn> = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Value(val) if urn.is_none() => {
+                    let val = val.to_string_lossy();
+
+                    if let Ok(val) = Urn::from_str(&val) {
+                        urn = Some(val);
+                    } else {
+                        return Err(anyhow!("invalid URN '{}'", val));
+                    }
+                }
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((Options { urn }, vec![]))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let profile = profile::default()?;
+    let signer = term::signer(&profile)?;
+    let storage = keys::storage(&profile, signer)?;
+
+    let urn = options
+        .urn
+        .or_else(|| radicle_common::project::cwd().ok().map(|(urn, _)| urn))
+        .ok_or_else(|| anyhow!("Couldn't get URN from either command line or cwd"))?;
+
+    let identity = any::get(&storage, &urn)?
+        .ok_or_else(|| anyhow!("No project or person found for this URN"))?;
+
+    let payload = match identity {
+        SomeIdentity::Project(_) => {
+            serde_json::to_string_pretty(project::verify(&storage, &urn)?.unwrap().payload())?
+        }
+        SomeIdentity::Person(_) => {
+            serde_json::to_string_pretty(person::verify(&storage, &urn)?.unwrap().payload())?
+        }
+        _ => todo!(),
+    };
+
+    if let Some(updated_payload) = term::Editor::new().edit(&payload).unwrap() {
```

Comment:
Mostly a stylistic thing, but would it not make more sense to have this code in the above match rather than matching on `identity` twice? It just comes at the cost of duplicating the `term::Editor` code.

Reply:
Agree that's preferrable to re-matching

Reply:
Made the changes [here](https://github.com/radicle-dev/radicle-cli/compare/ec0b9679b7a5e43fc48691fca49f262e317765bb..872886074161ff6e140725576f4ffffedb30e99b) but I find the one with re-match more readable.